examples/userapi/07_functions_on_subdomains.ipynb

```diff
@@ -0,0 +1,2999 @@
n+Va/0J/5mD0acxE/4zh7NnL/DkM69Ro3YXfAOaUaN2F57+92QuXvTsew9EnKYNIzxSiQl0fXx86NmzJwAff/xxjvMnTpxgx44dAPTt29fuXNbjJUuW5AhALBYLS5cuBaBfv3525/r06QNkbB+c26xu1jh69eqFt7e3o5ckLlKjehUAYn75jU1f576Vc0HeX/gZne/8FxcuxNOta2v+r0l99v50kMghY/nvzA9y1P/gozVEvfRfjp84Tb26YfTr04n/a1KfXbtjeOLJVxlw/0is+fwCu//BUYwbP4uqVSvQp3cHgoJKs3zFelrc/gA7v9tnVzcm5jDNb3uAhYs+w9fHh7t63EGPbm2pVrUiW7f9yMy3P8r1NeLjL9H6jkG8O28Zt4TXpvudbbBarSz+cDVt2j9CfHxijjbz5kdze9uHWb5iPZUrleeefl2oWyeMNV9s5c4eQ3nl1Xfs6j/ycG+aNK4PQJPG9Xnk4d62o3XrplSuFMIjD/emUqXyANzZ9Xa7OnVuKnwKj6P/RmfOnKVVm4eYNftjkpIuc1ePO2j2fzfzwYerua3NQOLjL+XyKs5JSMjo02QyERISbHfuwIEj/F/EPbz62rv8dfYCbVo3pXPHlpw99zcvT5hFm3Y5/00sFgv97vk3o16cxuHfTtLujgg6tm/Jnp9+pcXtD/DjngM4au0XW2nWYgALF3+OXylf+vXpRNP/C2fL1h+5/6FRDHl8fJ5t437/k1tb3senKzfSIqIhXTrfRmJiMrPnfMKdPYZpPXEp2bJWfHHpUdwXdf0rMakLAKNHjyY6OpoFCxbQv39/unXrBmRs8TtkyBDMZjP9+/cnPDzcrl1kZCSvvfYasbGxjBs3jtdee812bty4ccTGxhIaGsqgQYPs2jVo0IC7776bzz//nCFDhrBq1Sr8/DLuh/7yyy9ZuHAhRqORqKgoN195ISQl533OZILsOav51TUawa9U0eomX87706nBAP5+9nX987q33DF97u5ItWoVOXXqL7p2H5oRDHRoSbOmN9M8ogEVKuR3T3iGKVPn8/mnM7mrZzvbcwsXf86jj43jlVfnMvRf9+KX7Vrv7Ho7fXp3oGFD+3UQTp/+i569n+TTzzYRvWID997TNcdrnThxmqSky+zauYTGjTNmAM1mM889P5VZsz/mwYdf5GDMKtvX3tPf+oCEhEu8+spTvBT1uF1fly9fYdfumFyv6bNVX3Nn19vZ+s1CAgMzVhb5++8EOnV9jJ/2HWTO3KVEvfiYrf7PP8fy5NOTsFqtLHr/NR4e2Mt27st12+h7z7955dV3uP22/6NL59sAWDB/IhP+M4d9+w9xd+8OTHh5eI5xLJg/kQ6dH+XPP8/z4qghtG/XPNfxFsTRf6OnnpnEkSNxtG/XnM9WvGV7Dy5eTKB336f5bNXXRRpHfr74chsA3e5sbffh9/LlK9zd/xni4v7gpajHGffSUHx8Ms4nJ1/mX0Mn8MnSL3nu+am8P++fVR1mzfmENV9spUqVCny9/n/Ur5+xnofZbGbkC2/m+SEnL3/+eZ6Bj0Rx5UoKr77yFGNG/wtDZl7J7h9/4c4eQ1mwcCWtWjbiX0PuydF+wcKVRA66m3dmj7P9fMbF/cHtdwxk1+4Yolds4IH7ezg0JhFxzKFDh1i/fj0//vgjP/74I7/++itms5lXX32VsWPHFtxBHiwWCx988AEffPAB+/btIyEhgXLlynHzzTdzzz33MHx4zt/vnsBjZ3T37NlDq1atbEfWrmfvvvuu3fNnzpyxtWnWrBnTpk3DbDbTo0cPOnTowH333UedOnXYtGkT9evXZ+7cuTley9/fn2XLluHv78+kSZNo1KgRDzzwAI0aNWLSpEkEBASwfPlyWxCb3XvvvUfdunXZuHEjN910E/fddx8dOnSgZ8+emM1mZsyYQePG+S2idG0YyrbK8+C+EfaVq7XPu26vq36Q63bPu27HwfZ1m/TNu+5tV+1md/VjJ5Qu7c/GdfNo2aIRVquVzVt28fKEWdx195NUqtaeZs0HMPe9ZZjNed8y89STD9oFUACRg+4mvH4t4uMT2f3jL3bnmkc0zBHkAlStWpEpk58DIHrF+jxf76Wof9mCXMiYAZz6+giqVavIiROnWfHpBtu5v/7KSM3o0a1tjn78/EpxR9vcF9AKCPDj/Xmv2gI8gLJlA3lxVMbX2VfPfr816yPS09Ppe3dHuyAXoHu3tvzrsYzA583pC/O8Lndy5N/oxInTfLbqa4xGI3PeHmv3HgQHBzLn7bG2AM9ZZrOZo0d/Z/SY//Lxki8IC6vKW9NH29VZ9MEqjhyJ464ed/DqK0/ZglwAf38/3n1nPBUrluPDj9bw99//pADMnJURyI4f94QtyIWMn5c3Jmf8vDhi3vwVxMcncmuzW3gp6nG79yDi1gaMGf0vAN6ctijX9qGhlZg1c8w/ucdA9eqVeWp4xn/PG7/+3qHxiFxXPCR14Z133uGZZ55h0aJFxMTE5Pu3rbDi4+Pp0KEDkZGR7Nq1i4iICPr160fdunXZu3cv77//vtOv4S4eG+gmJCTw/fff245z5zJWT/z999/tns++/izAc889x4YNG7jzzjvZv38/n3/+OaVLlyYqKopdu3YREpL7Sq6tW7dm3759DBo0iAsXLrBixQouXLjAoEGD2LdvH61atcq1XcWKFdm9ezejR4+mdOnSfP755+zfv58777yTjRs38swzz7j2jZEiqV+/Fju3f8R3337EuJeGcmfX26lQISNn96d9Bxn+1ES63/UEqam5f7Xa66oAKsvN4Rk39Jw69VeOcykpqaxes5mXJ8xm2JOv8uhj4xg8ZCzvzctY9fJQ7PE8x/vIw71zPOfr68OAezO+pdi89Z+88OYRGSvXDn96Il+t/5YrV1JytM1NxK0NqFKlQqGvKSuP9pFBd+fa35DIjBSgbdv3uOQXq6Mc+Tfatn0PVquVZk1vJjw856rGDRvWteXTFsWJE6dtOcXef
```

Comment:
nitpicking: I'd use underscores to separate multi-words variable names, so `usave_grid` (applies to other variables as well)